sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1088,14 +1180,20 @@ private void releaseChannel(final Channel channel, final Promise<Void> promise)
      * {@link #acquire}.
      */
     private void runTasksInPendingAcquisitionQueue() {
+        ensureInEventLoop();
 
         this.ensureInEventLoop();
```

Comment:
Its very easy to slip stream changes violating the assumptions.
We need a mechanism to guard/detect (in order of preference) for these critical aspects. 

How about fail in our gates/CTL/perf runs at-least?

Reply:
its error trace and I am not seeing them at-least in my runs.

---
In reply to: [487800413](https://github.com/Azure/azure-sdk-for-java/pull/15157#discussion_r487800413) [](ancestors = 487800413)

Reply:
ACK - but I think we all agree that making the changes to fail fast will result in additional risk and need for validation (probably with changes to test infrastructure) - so IMO this is something we should do after releasing the hot fix. Makes sense?